### PR TITLE
Add support for QNX operating system in Yaml files

### PR DESCRIPTION
**Pull request by gang65**
Changelog: Feature: Add QNX operating system support
Docs: omit

### CODE REVIEW

File: conans/client/migrations_settings.py [hunk 1]
```diff
@@ -367,3 +367,82 @@
 """
 
 settings_1_15_1 = settings_1_15_0
+
```
Comment (lasote) [inline conans/client/migrations_settings.py]: Do not introduce changes in this module. We will do it once a version is released.

File: conans/client/conf/__init__.py [hunk 2]
```diff
@@ -47,6 +47,7 @@
     Arduino:
         board: ANY
     Emscripten:
+    QNX:
```
Comment (lasote) [inline conans/client/conf/__init__.py]: This is ok. But is that enough for managing the QNX operating system? Take a look at the AIX pull request changes. https://github.com/conan-io/conan/pull/4546
I'm totally ok adding QNX to the yml if you tell me no further changes are really required. but please take a look at it.

**Comment by lasote** [comment]
Take a look at the failing tests.

**Comment by lasote** [comment]
I think this PR https://github.com/conan-io/conan/pull/4297 is more advanced, could you go there and give feedback? Thanks!

File: conans/client/conf/__init__.py (hunk 2 above)
Comment (Johnnyxy) [reply conans/client/conf/__init__.py]: Hi,
at the moment I am using a customized ```settings.yml``` with:
```yaml
os:
    QNX:
        version: ["6.4", "6.5", "7.0"]
```

At the moment it is not forseeable if this is neccessary. But QNX is a real specialized system that one can configure down to the supported API etc. and I cannot safely say that without version differentiation everything will do fine in the future.

But as I and every QNX user have a customized ```settings.yml``` for this anyway, I would go with minimal additions to default Conan settings for the moment. We can add settings in the future.

**Pull request closed without merging**